File: SIC_Backend/Data/Models/UpdateDeviceModel.cs
namespace SIC_Backend.Data.Models
{
    public class UpdateDeviceModel
    {
        public required string Name { get; set; }
        public required DeviceDataTypes DataType { get; set; }
        public required int PlaceId { get; set; }
        public required int SectionId { get; set; }
        public required int Pin { get; set; }
    }
}

File: sic_home/lib/models/update_device_model.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sic_home/models/device.dart';

part 'update_device_model.g.dart';

@JsonSerializable()
class UpdateDeviceModel {
  String name;
  DeviceType deviceType;
  int placeId;
  int sectionId;
  int pin;

  UpdateDeviceModel({
    required this.name,
    required this.deviceType,
    required this.placeId,
    required this.sectionId,
    required this.pin,
  });

  factory UpdateDeviceModel.fromJson(Map<String, dynamic> json) =>
      _$UpdateDeviceModelFromJson(json);

  Map<String, dynamic> toJson() => _$UpdateDeviceModelToJson(this);
}
